pythonforandroid/bootstraps/common/build/build.py

```diff
@@ -321,16 +321,36 @@ def make_package(args):
                       'full private data into .apk.')
                 tar_dirs.append(args.private)
             else:
-                print('Copying main.py ONLY, since other app data is '
-                      'expected in site-packages.')
+                print("Copying main.py's ONLY, since other app data is "
+                      "expected in site-packages.")
                 main_py_only_dir = tempfile.mkdtemp()
                 _temp_dirs_to_clean.append(main_py_only_dir)
-                if exists(join(args.private, "main.pyo")):
-                    shutil.copyfile(join(args.private, "main.pyo"),
-                                    join(main_py_only_dir, "main.pyo"))
-                elif exists(join(args.private, "main.py")):
-                    shutil.copyfile(join(args.private, "main.py"),
-                                    join(main_py_only_dir, "main.py"))
+
+                # Check all main.py files we need to copy:
+                copy_paths = ["main.py", join("service", "main.py")]
+                for copy_path in copy_paths:
+                    variants = [
+                        copy_path,
+                        copy_path.partition(".")[0] + ".pyc",
+                        copy_path.partition(".")[0] + ".pyo",
+                    ]
+                    # Check in all variants with all possible endings:
+                    for variant in variants:
+                        if exists(join(args.private, variant)):
+                            # Make sure surrounding directly exists:
+                            dir_path = os.path.dirname(variant)
+                            if (len(dir_path) > 0 and
+                                    not exists(
+                                        join(main_py_only_dir, dir_path)
+                                    )):
+                                os.mkdir(join(main_py_only_dir, dir_path))
```

Comment:
Minor: What about creating the dir yes or yes? You know like `mkdir -p` so we can skip the if?
e.g.
```python
os.mkdir(join(main_py_only_dir, dir_path), exist_ok=True)
```

Reply:
I am trained to avoid this because of Python 2. Are we sure this code doesn't run in Python 2 when deploying for Python 2 android targets even with Python 3 host? It *might* be ok but I haven't tested just yet, and that's why I usually just avoid Python 3-only things

Reply:
Yes I see, let's play it safe then :+1:
Even though I don't really mind Python2 being broken :trollface:

Reply:
There's also `ensure_dir` in util.py, which I used to use for this.

Reply:
Hm my intuition is to stick more to standard functions for readability, especially since this will solve itself once we drop Python 2 entirely also as a target (which I assume we will do *eventually*?)